python/ray/air/result.py

```diff
@@ -53,6 +53,15 @@ def config(self) -> Optional[Dict[str, Any]]:
             return None
         return self.metrics.get("config", None)
 
+    @property
+    def uri(self) -> Optional[str]:
+        """Return checkpoint URI, if available.
+
+        Same as ``self.checkpoint.uri``."""
+        if not self.checkpoint:
+            return None
+        return self.checkpoint.uri
```

Comment:
Hm this actually doesn't make sense to me. The `Result` contains more than just the final checkpoint, so I don't think the checkpoint URI can be used as the Result URI as well. 

Seems we already have `log_dir`, which uniquely identifies the `Result`. Isn't that essentially the Result URI?

Reply:
The URI points to where the checkpoint is stored. This can be either a local directory or cloud path. `log_dir` is always a local directory.

Reply:
This is based on our discussion today - if we decide that this property is unnecessary in the end, it's also fine by me :)

Reply:
Yeah personally I would prefer this to return the Trial cloud path, and maybe we can also expose a public constructor that takes in the URI?